### ISSUE TITLE
TypeError when adding a product to wishlist without a referer

### ISSUE THREAD

**Issue by jurv** [BitBagCommerce/SyliusWishlistPlugin#237]
When calling the URL to add a product to the wishlist without having a referer, a TypeError exception is thrown :

```
Symfony\Component\HttpFoundation\Request::create(): Argument #1 ($uri) must be of type string, null given, called in [...]/vendor/bitbag/wishlist-plugin/src/Controller/Action/AddProductToWishlistAction.php on line 110
```

This happens when bots are browsing the site, but also when a visitor shares the "Add to wishlist" button URL to someone instead of the product page URL.

I thought this could be fixed by redirecting to the homepage if there is no referer:
```php
$referer = $request->headers->get('referer');
$redirectURL = $referer ? Request::create($referer)->getPathInfo() : $request->getSchemeAndHttpHost();

return new RedirectResponse($redirectURL);
```

Another option would be to inject the router in the controller and generate the URL to the product page.

I can submit a PR !

**Comment by DominikWolos** [comment]
Hi @jurv,
Could you describe some real-life situation, when this becomes an issue? I guess somebody would rather want to check the product first instead of adding it to wishlist instantly, especially when action is being performed not completely intentional.

I'd suggest, that sending URL performing action on users wishlist might not be well perceived by customer but maybe your case is different.

**Comment by jurv** [comment]
Well, the two cases I gave are real-life situations that we encountered in the last shop we deployed with this plugin : 

- bots : they are flooding our error logs. 
- users sharing the "Add to wishlist" links. I admit it's a weird case, but it actually happened.

You made a good point regarding the fact that performing an action on the wishlist without the user consent might not be well perceived. I didn't think of that. But I guess that made me question the behaviour of this controller even more 😅 Shouldn't this action only be accessible through a POST request then ? And a GET could redirect to the refered product's page, so that as you said the user could check out the product and not have it automatically added to his wishlist without maybe realising it ?
